### PR TITLE
Play 2.6 updates

### PR DESCRIPTION
**Pull request by TimMoore**
This fixes compatibility with Play 2.6.

Follow up from #60.

It might be easier to review each commit individually, as each one contains one logical change.

We'll probably need to make similar changes to the Java version.

### CODE REVIEW

**Comment by TimMoore** [comment]
@gmethvin, @marcospereira, @richdougherty I requested reviews from you because it will probably be good to have a Play developer check to be sure we're representing best practices in this example. You don't all need to review; one Play reviewer is probably enough.

**Comment by TimMoore** [comment]
Once this is merged, I can go ahead and make the changes on the Java side.

**Comment by beritou** [comment]
> CC: @beritou ... I made the change to use hidden form fields instead of URL rewriting for CSRF tokens. It seems to work well. Let me know what you think.

Looks great. I wish I had thought of making the change inside of `foundationForm` like that 😉

File: web-gateway/app/controllers/AbstractAuctionController.scala [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 import org.ocpsoft.prettytime.units.JustNow
-import play.api.i18n.{Messages, MessagesApi}
-import play.api.mvc.{Controller, RequestHeader, Result}
+import play.api.i18n.Messages
+import play.api.mvc._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ ExecutionContext, Future }
+
+abstract class AbstractAuctionController(userService: UserService, controllerComponents: ControllerComponents)
```
Comment (marcospereira) [inline web-gateway/app/controllers/AbstractAuctionController.scala]: Hum... maybe we should make the [`defaultExecutionContext` implicit](https://github.com/playframework/playframework/blob/10dfaea2f6393dec1e0e40056d93662a3cf54c06/framework/src/play/src/main/scala/play/api/mvc/Controller.scala#L78), so you won't need to inject it all over the place. Can't remember right now if we decide not to make it implicit or if it is just a detail that slipped.

File: web-gateway/app/controllers/Main.scala [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 import play.api.data.Form
 import play.api.data.Forms._
-import play.api.i18n.MessagesApi
-import play.api.mvc.Action
+import play.api.mvc.ControllerComponents
 
 import scala.concurrent.ExecutionContext
 
-class Main(messagesApi: MessagesApi, userService: UserService)(implicit ec: ExecutionContext) extends AbstractController(messagesApi, userService) {
+class Main(userService: UserService, controllerComponents: ControllerComponents)
```
Comment (marcospereira) [inline web-gateway/app/controllers/Main.scala]: Nitpicking: we are moving away from this `Application` (or in this case `Main`) naming scheme to use `SomethingController`. Maybe `HomeController` here.

File: web-gateway/conf/application.conf [hunk 3]
```diff
@@ -3,7 +3,4 @@
 lagom.circuit-breaker.default.call-timeout = 5s
 
 play.application.loader = WebGatewayLoader
-
-play.crypto.secret = "somesecret"
```
Comment (marcospereira) [inline web-gateway/conf/application.conf]: 👍

File: web-gateway/app/controllers/AbstractAuctionController.scala (hunk 1 above)
Comment (TimMoore) [reply web-gateway/app/controllers/AbstractAuctionController.scala]: From the doc comment, it looks like it was intentional, but it's not clear why.

In any case, we can do it here for sure.

**Comment by TimMoore** [comment]
Thanks for the reviews @renatocaval and @marcospereira.

Marcos, I'm going to go ahead and merge this now so that the example is working on `master`, and then I'll address your review comments in a follow-up pull request (along with a few other minor changes I'd like to make).

**Pull request merged**